source/extensions/extensions_build_config.bzl

```diff
@@ -71,6 +71,7 @@ EXTENSIONS = {
     "envoy.filters.http.health_check":                  "//source/extensions/filters/http/health_check:config",
     "envoy.filters.http.ip_tagging":                    "//source/extensions/filters/http/ip_tagging:config",
     "envoy.filters.http.jwt_authn":                     "//source/extensions/filters/http/jwt_authn:config",
+    "envoy.filters.http.kill_request":                  "//source/extensions/filters/http/kill_request:kill_request_config",
```

Comment:
So, you include here but then exclude later.. how does a user actually include this in a build? Can you comment on this to make it clearer?

Reply:
Thanks Harvey. Added comment.

Reply:
Do you mean `all_extensions.bzl` in the comment? I still thinking adding it and then blocking by default is a bit weird. 

Can we do this a bit differently and add a new set of definitions here, DISABLED_BY_DEFAULT_EXTENSIONS, that looks like the current structure but only has the kill filter. We can feed this definition into both all_extensions.bzl and also into the docs.

Reply:
Thanks Harvey! Added the filter to DISABLED_BY_DEFAULT_EXTENSIONS.

Reply:
This is much better. What I was hinting at ^^ is that we should have `DISABLED_BY_DEFAULT_EXTENSIONS` as you've done, but also not have to include the kill request filter in `EXTENSIONS`. This should be easy to do; you might need to modify some consumers, e.g. doc build, to consume from `DISABLED_BY_DEFAULT_EXTENSIONS`. That way, you can tell folks to move their extension from `DISABLED_BY_DEFAULT_EXTENSIONS` to `EXTENSIONS` to enable.

Reply:
ah :) thank you Harvey for the explanation! Removed the filter from `EXTENSIONS` and updated the consumers to make the CI doc test pass. PTAL